src/main/java/frc/robot/subsystems/Climber_Subsystem.java

```diff
@@ -27,29 +28,100 @@
+        verticalL.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 50);
+        horizontalR.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 50);
+        horizontalL.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 50);
+
+    }
+    
+    //Initialization function that should be called only once
+    //This should be called before climbing but not on robot boot so manual adjustments to the climber
+    //before the game starts won't mess it up
+    public void singleInit()
+    {
+        //Hard reset the encoders to 0
+        verticalR.setSelectedSensorPosition(0, 0, 50);
+        verticalL.setSelectedSensorPosition(0, 0, 50);
+        horizontalR.setSelectedSensorPosition(0, 0, 50);
+        horizontalL.setSelectedSensorPosition(0, 0, 50);
+
+        // NOTE: Forward soft limit _must_ be the higher number. It doesn't really matter about which direction it goes
+        // soft limits are a range. Reverse is the low number and forward is the high number
+        int horizontalReverseLimit = 100;
+        int horizontalForwardLimit = 12000;
+        int verticalReverseLimit = 1000;
+        int verticalForwardLimit = 21000;
+        
+        
+        SmartDashboard.putNumber("Horizontal Forward Limit", horizontalForwardLimit);
+        SmartDashboard.putNumber("Horizontal Reverse Limit", horizontalReverseLimit);
+        SmartDashboard.putNumber("Vertical Forward Limit", verticalForwardLimit);
+        SmartDashboard.putNumber("Vertical Reverse Limit", verticalReverseLimit);
+
+        verticalR.configForwardSoftLimitThreshold(verticalForwardLimit);
+        verticalR.configReverseSoftLimitThreshold(verticalReverseLimit);
+        verticalL.configForwardSoftLimitThreshold(verticalForwardLimit);
+        verticalL.configReverseSoftLimitThreshold(verticalReverseLimit);
+
+        horizontalR.configForwardSoftLimitThreshold(horizontalForwardLimit);
+        horizontalR.configReverseSoftLimitThreshold(horizontalReverseLimit);
+        horizontalL.configForwardSoftLimitThreshold(horizontalForwardLimit);
+        horizontalL.configReverseSoftLimitThreshold(horizontalReverseLimit);
+
+
+        verticalR.configForwardSoftLimitEnable(true);
+        verticalR.configReverseSoftLimitEnable(true);
+        verticalL.configForwardSoftLimitEnable(true);
+        verticalL.configReverseSoftLimitEnable(true);
 
-        verticalR.setSelectedSensorPosition(0, 0, 10);
-        verticalL.setSelectedSensorPosition(0, 0, 10);
-        horizontalR.setSelectedSensorPosition(0, 0, 10);
-        horizontalL.setSelectedSensorPosition(0, 0, 10);
+        horizontalL.configForwardSoftLimitEnable(true);
+        horizontalL.configReverseSoftLimitEnable(true);
+        horizontalR.configForwardSoftLimitEnable(true);
+        horizontalR.configReverseSoftLimitEnable(true);
+    }
 
+    public void updateState()
```

Comment:
This is the thing that resets up the talons according to the smart dashboard's choosers.

It first makes sure that you've chosen _something_ for the `m_stateFlag` chooser. If you haven't it will do nothing.

Then if either it's never been initialized (`updateState == null`) or the value of the state flag has changed `updateState != Robot.m_stateFlag.getSelected()` it will reset itself according to the latest from the SmartDashboard.

Finally it records the current state of `m_stateFlag` into `updateState` so next time through it can tell if it changed.


Ideally this would just monitor all 8 of the different choosers and change them individually whenever something changed. I did it this way because it was faster. It should be fine for testing.